sdk/ml/azure-ai-ml/documentation_guidelines.md

```diff
@@ -0,0 +1,103 @@
+## **AzureML v2 Reference Documentation Guide**
+
+When updating the codebase, please note the following guidelines for creating user-friendly docstrings.
+ 
+#### Principles 
+
+1. **Accuracy**: The documentation accurately reflects the features and functionality of the product.
+2. **Completeness**: The documentation covers all relevant features and functionality of the product.
+3. **Clarity**: The documentation is written in clear, concise language that is easy to understand.  
+4. **Consistency**: The documentation has a consistent format and structure, making it easy to navigate and follow.  
+5. **Relevance**: The documentation is up-to-date and relevant to the current version of the product. 
+6. **Coverage**: Every public object must have a docstring. Docstrings are encouraged, but not required for private objects. 
+7. **Demonstration**: Every docstring should include an up-to-date code snippet that demonstrates how to use the product effectively. 
+
+
+##### Accuracy 
+
+- No broken links are allowed in reference documentation, and all links must be location/language neutral (e.g. no /en-us/ in links)
+- Docstring details should be aligned with type hints 
+- Any code snippets should be defined in the samples directory in this folder so that they are tested during every CI run
+- Constructors (`def __init__`) should return None, per [PEP 484 standards](https://peps.python.org/pep-0484/#the-meaning-of-annotations)
+
+ 
+##### Completeness 
+
+- Be sure to warn of any edge cases or note special circumstances, if applicable. This can be done using an admonition in the docstring. In addition to the “tip” class we use for examples, Sphinx provides classes for other types of messages that you should feel free to use if appropriate: 
```

Comment:
Nit: Taking this chance to explicitly state what sphinx is for people who are unfamiliar with it:
```suggestion
- Be sure to warn of any edge cases or note special circumstances, if applicable. This can be done using an admonition in the docstring. In addition to the “tip” class we use for examples, Sphinx (the python docs generator that we use) provides classes for other types of messages that you should feel free to use if appropriate: 
```

Edit: Nevermind, you do that below.